### PR TITLE
Fixed opnorm

### PR DESCRIPTION
**Pull request by andreasnoack**
A rebased version of https://github.com/JuliaLang/julia/pull/30138

### CODE REVIEW

File: stdlib/LinearAlgebra/src/diagonal.jl [hunk 1]
```diff
@@ -1036,6 +1036,16 @@ end
...
 
+_opnorm12Inf(A::Diagonal, p) = maximum(opnorm(x, p) for x in A.diag)
+
+function opnorm(A::Diagonal, p::Real=2)
+    if p == 1 || p == Inf || p == 2
+        return _opnorm12Inf(A, p)
+    else
+        throw(ArgumentError("invalid p-norm p=$p. Valid: 1, 2, Inf"))
+    end
+end
```
Comment (andreasnoack) [inline stdlib/LinearAlgebra/src/diagonal.jl:1047]:
```suggestion
end
```

Comment (andreasnoack) [reply stdlib/LinearAlgebra/src/diagonal.jl:1047]: @dgleich would you be able to accept this change? I can't since the PR is from your fork of Julia.

Comment (dgleich) [reply stdlib/LinearAlgebra/src/diagonal.jl]: Is it just the one line ending? If so, I pushed a change.

**Comment by andreasnoack** [comment]
Is CI currently broken? The failures look unrelated to any changes here.

File: stdlib/LinearAlgebra/src/diagonal.jl (hunk 1 above)
Comment (andreasnoack) [reply stdlib/LinearAlgebra/src/diagonal.jl]: Yes. It was just the white space here.

**Comment by DilumAluthge** [comment]
I restarted the failed job - let's see if the retry passes.

**Comment by DilumAluthge** [comment]
Hmmm. The failure persists after a retry.

**Comment by DilumAluthge** [comment]
Keno suggests that this has been fixed on master, so perhaps we need to rebase this PR again?

**Comment by DilumAluthge** [comment]
FYI, the PowerPC jobs will likely fail here. This is a known issue, and is not related to this PR.

**Comment by andreasnoack** [comment]
@dgleich would you be able to rebase the branch? I tried but there are conflicts.

**Comment by DilumAluthge** [comment]
Bump @dgleich ^

**Comment by dgleich** [comment]
Done. Thanks for the bump. Something went wrong with one of the automatic rebase steps, so I had to fix it with another commit. (But I imagine these get squashed when merged?)

**Comment by andreasnoack** [comment]
@dkarrasch I was about to review this but I'm formally the author of the PR although I didn't write a line of it so I'm not allowed to review. Would you be able to?

**Comment by dgleich** [comment]
@andreasnoack do you want me to send in a new pull request from my account for this one so you can do the review?

**Comment by dkarrasch** [comment]
@dgleich Indeed, you could actually prepare a new PR to the now outsourced https://github.com/JuliaLang/LinearAlgebra.jl. I'll try to find some time to make a review (though it's not technically necessary for merging).

**Comment by KristofferC** [comment]
Moved to https://github.com/JuliaLang/LinearAlgebra.jl/pull/1138

**Pull request closed without merging**